src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -0,0 +1,506 @@
+     strings::driver_braking},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeWiperStatus,
+     strings::wiper_status},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeHeadLampStatus,
+     strings::head_lamp_status},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeEngineTorque,
+     strings::engine_torque},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeAccPedalPosition,
+     strings::acc_pedal_pos},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeSteeringWheelAngle,
+     strings::steering_wheel_angle},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeECallInfo,
+     strings::e_call_info},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeAirbagStatus,
+     strings::airbag_status},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeEmergencyEvent,
+     strings::emergency_event},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeClusterModeStatus,
+     strings::cluster_mode_status},
+    {hmi_apis::FunctionID::VehicleInfo_SubscribeMyKey, strings::my_key},
+};
+}
+#endif  // #ifdef HMI_DBUS_API
+
+void SubscribeVehicleDataRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "NULL pointer");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+  std::string info;
+  mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  smart_objects::SmartObject response_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  bool result = false;
+  CheckVISubscriptions(
+      app, info, result_code, response_params, msg_params, result);
+
+  if (mobile_apis::Result::INVALID_ENUM != result_code) {
+    SendResponse(result,
+                 result_code,
+                 info.empty() ? NULL : info.c_str(),
+                 response_params.empty() ? NULL : &response_params);
+    return;
+  }
+
+#ifdef HMI_DBUS_API
+  // Generate list of subrequests
+  for (size_t i = 0; i < sizeof(subrequests) / sizeof(subrequests[0]); ++i) {
+    const Subrequest& sr = subrequests[i];
+    if (true == (*message_)[strings::msg_params].keyExists(sr.str) &&
+        true == (*message_)[strings::msg_params][sr.str].asBool()) {
+      HmiRequest hmi_request;
+      hmi_request.str = sr.str;
+      hmi_request.func_id = sr.func_id;
+      hmi_request.complete = false;
+      hmi_requests_.push_back(hmi_request);
+    }
+  }
+  LOG4CXX_DEBUG(
+      logger_, hmi_requests_.size() << " requests are going to be sent to HMI");
+
+  // Send subrequests
+  for (HmiRequests::const_iterator it = hmi_requests_.begin();
+       it != hmi_requests_.end();
+       ++it)
+    SendHMIRequest(it->func_id, &msg_params, true);
+#else
+  StartAwaitForInterface(HmiInterfaces::HMI_INTERFACE_VehicleInfo);
+  SendHMIRequest(hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData,
```

Comment:
cannot see the code to process VehicleInfo_SubscribeVehicleData response in onEvnet() like other requests.